tests/model_serving/model_runtime/triton/basic_model_deployment/utils.py

```diff
@@ -0,0 +1,173 @@
+
+        try:
+            proc = subprocess.run(
+                args=args,
+                capture_output=True,
+                text=True,
+                check=True,
+            )
+            return json.loads(proc.stdout)
+        except subprocess.CalledProcessError as e:
+            return f"gRPC request (inline) failed:\nSTDOUT: {e.stdout}\nSTDERR: {e.stderr}"
+
+
+def run_triton_inference(
+    pod_name: str, isvc: InferenceService, input_data: dict[str, Any], model_name: str, protocol: str, root_dir: str
+) -> Any:
+    deployment_mode = isvc.instance.metadata.annotations.get("serving.kserve.io/deploymentMode")
+    rest_endpoint = f"/v2/models/{model_name}/infer"
+
+    if protocol not in (Protocols.REST, Protocols.GRPC):
+        return f"Invalid protocol {protocol}"
+
+    is_rest = protocol == Protocols.REST
+
+    if deployment_mode == KServeDeploymentType.RAW_DEPLOYMENT:
+        port = TRITON_REST_PORT if is_rest else TRITON_GRPC_PORT
+        with portforward.forward(pod_or_service=pod_name, namespace=isvc.namespace, from_port=port, to_port=port):
+            host = f"{LOCAL_HOST_URL}:{port}" if is_rest else get_grpc_url(base_url=LOCAL_HOST_URL, port=port)
+            return (
+                send_rest_request(f"{host}{rest_endpoint}", input_data)
+                if is_rest
+                else send_grpc_request(host, input_data, root_dir)
+            )
+
+    elif deployment_mode == KServeDeploymentType.SERVERLESS:
+        base_url = isvc.instance.status.url.rstrip("/")
+        if is_rest:
+            return send_rest_request(f"{base_url}{rest_endpoint}", input_data)
+        else:
+            grpc_url = get_grpc_url(base_url=base_url, port=TRITON_GRPC_REMOTE_PORT)
+            return send_grpc_request(grpc_url, input_data, root_dir, insecure=True)
+
+    return f"Invalid deployment_mode {deployment_mode}"
+
+
+def get_grpc_url(base_url: str, port: int) -> str:
+    return f"{base_url.replace('https://', '').replace('http://', '')}:{port}"
+
+
+def validate_inference_request(
+    pod_name: str,
+    isvc: InferenceService,
+    response_snapshot: Any,
+    input_query: Any,
+    model_name: str,
+    protocol: str,
+    root_dir: str,
+) -> None:
+    response = run_triton_inference(
+        pod_name=pod_name,
+        isvc=isvc,
+        input_data=input_query,
+        model_name=model_name,
+        protocol=protocol,
+        root_dir=root_dir,
+    )
+    assert response == response_snapshot, f"Output mismatch: {response} != {response_snapshot}"
+
+
+def get_gpu_identifier(accelerator_type: str) -> str:
+    return ACCELERATOR_IDENTIFIER.get(accelerator_type.lower(), Labels.Nvidia.NVIDIA_COM_GPU)
+
+
+def get_template_name(protocol: str, accelerator_type: str) -> str:
+    """
+    Returns template name based on protocol and accelerator type.
+    Falls back to default TRITON_REST if not found.
+    """
+    key = f"{protocol.lower()}_{accelerator_type.lower()}"
```

Comment:
do you want to  check if template is present or now ?

Reply:
Triton is under Tested & Verified runtime. We don't deliver its Template as a part of RHOAI. So When we execute these tests, template would never be exists. We always have to create template first.